framework/src/packets/icmp/v6/ndp/options/link_layer_addr.rs

```diff
@@ -88,6 +118,15 @@ impl LinkLayerAddress {
     pub fn set_addr(&mut self, addr: MacAddr) {
         self.fields_mut().addr = addr;
     }
+
+    #[inline]
+    pub fn set_option_type(&mut self, option_type: u8) {
+        if option_type == SOURCE_LINK_LAYER_ADDR || option_type == TARGET_LINK_LAYER_ADDR {
+            self.fields_mut().option_type = option_type
+        } else {
+            panic!("Option type needs to be either source link layer(1) or target link layer address(2)")
```

Comment:
don't panic, that crashes the whole app. let's revert this back to the previous version without checks. and come back to this later.

Reply:
removed the panic but i left in the check. i added a todo so that we can address what to do at some point